src/macros.jl

```diff
@@ -801,7 +819,7 @@ macro defVar(args...)
     # We now build the code to generate the variables (and possibly the JuMPDict
     # to contain them)
     refcall, idxvars, idxsets, idxpairs, condition = buildrefsets(var)
-    code = :( $(refcall) = Variable($m, $lb, $ub, $(quot(t)), "", $value) )
+    code = :( $(refcall) = $kernelfunc($m, $lb, $ub, $(quot(t)), "", $value) )
     if symmetric
```

Comment:
Yeah, there should be some mechanism for a nice fallback error message in the case of an unrecognized kwarg or if you only implement a few of the ``Variable`` methods implemented or something.